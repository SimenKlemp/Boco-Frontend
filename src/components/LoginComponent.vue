<template>
  <div class="loginContainer">
    <div class="titleInfo">
      <h1 id="loginTitle">Velkommen!</h1>
    </div>
    <div>
      <h5 id="LoginError" v-if="this.$store.state.userInfo.status === 204">
        Feil brukernavn eller passord!
      </h5>
    </div>
    <div id="RegContainer">
      <form @submit.prevent="submit">
        <div id="registerTitle">
          <label>Logg inn:</label>
        </div>

        <BaseInput
          id="email"
          class="mb-4"
          type="email"
          v-model="email"
          placeholder="Epost"
        />
        <BaseInput
          id="password"
          class="mb-4"
          type="password"
          v-model="password"
          placeholder="Passord"
        />

        <BaseButton v-on:click="doLogin" text="Sign in" />

        <div v-if="this.$store.state.userInfo.status === 204">
          <router-link to="/registration"
            >If you do not have a user already, click here to
            register</router-link
          >
        </div>
      </form>
    </div>
  </div>
</template>

<script>
import BaseInput from "@/components/baseTools/BaseInput";
import BaseButton from "@/components/baseTools/BaseButton";
import { doLogin } from "@/service/apiService";

export default {
  name: "LoginComponent",
  components: {
    BaseInput,
    BaseButton,
  },
  data() {
    return {
      email: "",
      password: "",
    };
  },
  methods: {
    async submit() {
      let loginResponse = await doLogin(this.email, this.password);

      if (loginResponse.status === 200) {
        this.$store.dispatch("storeUser", loginResponse.data.userInfo);
        this.$store.dispatch("storeToken", loginResponse.data.token);

        switch (loginResponse.data.userInfo.role) {
          case "USER":
            //TODO: push til min side!! -->
            this.$router.push({ name: "HomeView" });
            break;

          default:
            alert("Something went wrong with the authentication!");
        }
      } else {
        this.$store.dispatch("storeUser", loginResponse);
      }
    },
  },
};
</script>

<style scoped>
.loginContainer {
  display: flex;
  flex-direction: column;
  row-gap: 2px;
  justify-content: center;
  margin: 50px auto;
  margin-top: 15%;
  width: 300px;
}

#registerTitle {
  font-size: x-large;
  font-weight: bold;
}
#loginTitle {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial,
    sans-serif, "Apple Color Emoji", "Segoe UI Emoji";
  font-weight: 100;
  font-size: 28px;
  color: black;
}
#LoginError {
  color: black;
  font-size: 15px;
  font-weight: 300;
  padding: 8px;
  margin: 0 auto;
  margin-bottom: 5px;
  border-radius: 5px;
  border: solid 1px red;
  background-color: rgba(255, 0, 0, 0.27);
}
form {
  padding: 0px 30px 0px 30px;
}
form > * {
  margin-bottom: 10px;
}
</style>
