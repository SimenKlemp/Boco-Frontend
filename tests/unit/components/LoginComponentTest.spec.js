import LoginComponent from "@/components/LoginComponent";
import { mount } from "@vue/test-utils";

describe("LoginComponent", () => {
  let $store;
  beforeEach(() => {
    $store = {
      commit: () => {},
      dispatch: () => {},
      state: {
        userInfo: "",
      },
    };
  });
  test("If the page renders", () => {
    const wrapper = mount(LoginComponent, {
      global: {
        mocks: {
          $store,
        },
      },
    });

    expect(wrapper.login).toMatchSnapshot();
  });
  test("When login button is clicked", () => {});
});
