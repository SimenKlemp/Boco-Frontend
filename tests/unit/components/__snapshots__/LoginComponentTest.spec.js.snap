// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent If the page renders 1`] = `undefined`;
